alfresco-solrapi-client/alfresco-solrapi-client-spring/src/test/java/eu.xenit.alfresco.solrapi.client.spring/SpringModelMapperTest.java

```diff
@@ -0,0 +1,46 @@
+package eu.xenit.alfresco.solrapi.client.spring;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatExceptionOfType;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+import eu.xenit.alfresco.client.solrapi.api.model.ChildAssociation;
+import org.junit.jupiter.api.Test;
+
+class SpringModelMapperTest {
+
+    private SpringModelMapper mapper = new SpringModelMapper();
+
+    @Test
+    void createChildAssociation() {
+
+        // Note: forward slashes are escaped
+        String childAssocStr = "workspace://SpacesStore/4c5cfb80-111e-4fc4-9176-e8eaefcaf632"
+                + "|workspace://SpacesStore/0a49ddb4-214b-41e7-92c9-3cc5e1960148"
+                + "|{http://www.alfresco.org/model/content/1.0}contains"
+                + "|{http://www.alfresco.org/model/application/1.0}dictionary"
+                + "|true|-1";
+
+        ChildAssociation assoc = this.mapper.createChildAssociation(childAssocStr);
+
+        assertThat(assoc).isNotNull();
+        assertThat(assoc.getParentRef()).isEqualTo("workspace://SpacesStore/4c5cfb80-111e-4fc4-9176-e8eaefcaf632");
+        assertThat(assoc.getChildRef()).isEqualTo("workspace://SpacesStore/0a49ddb4-214b-41e7-92c9-3cc5e1960148");
+        assertThat(assoc.getAssocTypeQName()).isEqualTo("{http://www.alfresco.org/model/content/1.0}contains");
+        assertThat(assoc.getAssocTypeQName()).isEqualTo("{http://www.alfresco.org/model/application/1.0}dictionary");
```

Comment:
failing build -> `assoc.getAssocQName()`

Reply:
How did I manage to screw that up ? :-)